pkg/kubelet/kubelet_pods.go

```diff
@@ -224,13 +225,35 @@ func makeHostsMount(podDir, podIP, hostName, hostDomainName string, hostAliases
 
 // ensureHostsFile ensures that the given host file has an up-to-date ip, host
 // name, and domain name.
-func ensureHostsFile(fileName, hostIP, hostName, hostDomainName string, hostAliases []v1.HostAlias) error {
-	content := hostsFileContent(hostIP, hostName, hostDomainName, hostAliases)
-	return ioutil.WriteFile(fileName, content, 0644)
+func ensureHostsFile(fileName, hostIP, hostName, hostDomainName string, hostAliases []v1.HostAlias, useHostNetwork bool) error {
+	var hostsFileContent []byte
+	var err error
+
+	if useHostNetwork {
+		// if Pod is using host network, read hosts file from the node's filesystem.
+		hostsFileContent, err = nodeHostsFileContent()
+	} else {
+		// if Pod is not using host network, create a managed hosts file with Pod IP and other information.
+		hostsFileContent, err = managedHostsFileContent(hostIP, hostName, hostDomainName, hostAliases)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return ioutil.WriteFile(fileName, hostsFileContent, 0644)
+}
+
+// nodeHostsFileContent reads the content of node's hosts file.
+func nodeHostsFileContent() ([]byte, error) {
+	// `etcHostsPath` references the location of the hosts file on the node.
+	// `/etc/hosts` for *nix systems.
+	return ioutil.ReadFile(etcHostsPath)
 }
 
-// hostsFileContent is the content of the managed etc hosts
-func hostsFileContent(hostIP, hostName, hostDomainName string, hostAliases []v1.HostAlias) []byte {
+// managedHostsFileContent generates the content of the managed etc hosts based on Pod IP and other
+// information.
+func managedHostsFileContent(hostIP, hostName, hostDomainName string, hostAliases []v1.HostAlias) ([]byte, error) {
```

Comment:
Don't think returning the error is necessary since it's always nil. I think it's ok to not add it.